### PR TITLE
SB-27532 feat: Private APIs for enrolment read and list

### PR DESCRIPTION
**Pull request by PradyumnaNagendra**
<table>
<td>
  <a href="https://project-sunbird.atlassian.net/browse/SB-27532" title="SB-27532" target="_blank">EnhancementSB-27532</a>  Validating authToken for enrol API
  </td></table>
  <br />
 

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Enhancement (additive changes to improve performance)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions:
* Hardware versions:

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: ansible/roles/kong-api/defaults/main.yml [hunk 1]
```diff
@@ -1535,6 +1535,24 @@ kong_apis:
...
+    uris: "{{ course_service_prefix }}/private/v1/user/enrollment/list"
+    upstream_url: "{{ lms_service_url }}/private/v1/user/courses/list"
+    strip_uri: true
+    plugins:
+    - name: jwt
+    - name: cors
+    - "{{ statsd_pulgin }}"
+    - name: acl
+      config.whitelist:
+        - 'anonymousCourseAccess'
```
Comment (keshavprasadms) [inline ansible/roles/kong-api/defaults/main.yml:1548]: So this API can be called without logging in or X-Auth-Token?

File: ansible/roles/kong-api/defaults/main.yml [hunk 2]
```diff
@@ -1553,6 +1571,24 @@ kong_apis:
...
+    uris: "{{ course_service_prefix }}/private/v2/user/enrollment/list"
+    upstream_url: "{{ lms_service_url }}/private/v2/user/courses/list"
+    strip_uri: true
+    plugins:
+    - name: jwt
+    - name: cors
+    - "{{ statsd_pulgin }}"
+    - name: acl
+      config.whitelist:
+        - 'anonymousCourseAccess'
```
Comment (keshavprasadms) [inline ansible/roles/kong-api/defaults/main.yml:1584]: So this API can be called without logging in or X-Auth-Token?

Comment (PradyumnaNagendra) [reply ansible/roles/kong-api/defaults/main.yml:1584]: Yes, these are private APIs for internal purpose

File: ansible/roles/kong-api/defaults/main.yml (hunk 1 above)
Comment (PradyumnaNagendra) [reply ansible/roles/kong-api/defaults/main.yml:1548]: Yes, these are private APIs for internal purpose

File: ansible/roles/kong-api/defaults/main.yml (hunk 2 above)
Comment (PradyumnaNagendra) [reply ansible/roles/kong-api/defaults/main.yml]: As discussed, I have added the new ACL and updated APIs list also

File: ansible/roles/kong-api/defaults/main.yml (hunk 1 above)
Comment (PradyumnaNagendra) [reply ansible/roles/kong-api/defaults/main.yml]: I have added the new ACL and updated APIs list also

**Review by keshavprasadms — APPROVED** [review]

**Pull request merged**